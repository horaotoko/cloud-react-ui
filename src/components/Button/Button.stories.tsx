import { text, radios, boolean } from "@storybook/addon-knobs";
import { ButtonContainer, ButtonProps } from "./ButtonContainer";

export default {
  component: ButtonContainer,
  title: "components/Button",
};

const sizeOptions: { [key: string]: ButtonProps["size"] } = {
  small: "small",
  medium: "medium",
  large: "large",
};

const colorOptions: { [key: string]: ButtonProps["color"] } = {
  default: "default",
  danger: "danger",
  primary: "primary",
  settings: "settings",
};

const iconOptions: { [key: string]: ButtonProps["icon"] } = {
  none: undefined,
  plus: "plus",
};

const iconPlacementOptions: { [key: string]: ButtonProps["iconPlacement"] } = {
  start: "start",
  end: "end",
};

export const Button: React.FC = () => (
  <ButtonContainer
    size={radios("Size", sizeOptions, "small") as ButtonProps["size"]}
    color={radios("Color", colorOptions, "default") as ButtonProps["color"]}
    disabled={boolean("Disabled", false)}
    icon={radios("Icon", iconOptions, undefined) as ButtonProps["icon"]}
    iconPlacement={
      radios(
        "IconPlacement",
        iconPlacementOptions,
        undefined
      ) as ButtonProps["iconPlacement"]
    }
    onClick={() => console.log("button clicked")}
  >
    {text("Button Text", "Button")}
  </ButtonContainer>
);
