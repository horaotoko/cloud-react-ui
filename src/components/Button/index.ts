export { ButtonContainer as Button } from "./ButtonContainer";
export type { ButtonProps } from "./ButtonContainer";
