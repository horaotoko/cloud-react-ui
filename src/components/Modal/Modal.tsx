import { forwardRef } from "react";
import styled, { css } from "styled-components";
import {
  Dialog as MuiDialog,
  DialogProps as MuiDialogProp,
} from "@material-ui/core";
import { defaultProps } from "../../theme";

export type Props = MuiDialogProp;

const StyledModal = styled(MuiDialog)`
  ${({ theme: { modal } }) => css`
    color: ${modal.color};
  `}
`;
StyledModal.defaultProps = defaultProps;

const Modal = forwardRef<HTMLDivElement, Props>(
  ({ children, ...rest }, ref) => (
    <StyledModal ref={ref} {...rest}>
      {children}
    </StyledModal>
  )
);

Modal.displayName = "Modal";

export { Modal };
