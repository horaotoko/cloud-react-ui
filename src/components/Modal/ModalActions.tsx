import { forwardRef } from "react";
import styled, { css } from "styled-components";
import { defaultProps } from "../../theme";

export type ModalActionsProps = {
  children?: React.ReactNode;
  className?: string;
};

const StyledModalActions = styled.div`
  ${({ theme: { modal } }) => css`
    display: ${modal.actions.display};
    align-items: ${modal.actions.alignItems};
    justify-content: ${modal.actions.justifyContent};
    flex: ${modal.actions.flex};
    padding: ${modal.actions.padding};
    border-top-style: ${modal.actions.borderTopStyle};
    border-top-width: ${modal.actions.borderTopWidth};
    border-top-color: ${modal.actions.borderTopColor};
    & > * {
      margin: ${modal.actions.children.margin};
    }
  `}
`;
StyledModalActions.defaultProps = defaultProps;

const ModalActions = forwardRef<HTMLDivElement, ModalActionsProps>(
  (props, ref) => <StyledModalActions ref={ref} {...props} />
);

ModalActions.displayName = "Modal.Actions";

export { ModalActions };
