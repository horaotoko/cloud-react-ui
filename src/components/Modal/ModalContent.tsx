import { forwardRef } from "react";
import styled, { css } from "styled-components";
import { defaultProps } from "../../theme";

export type ModalContentProps = {
  children?: React.ReactNode;
  className?: string;
};

const StyledModalContent = styled.div`
  ${({ theme: { modal } }) => css`
    flex: ${modal.content.flex};
    padding: ${modal.content.padding};
    overflow-y: ${modal.content.overflowY};
  `}
`;
StyledModalContent.defaultProps = defaultProps;

const ModalContent = forwardRef<HTMLDivElement, ModalContentProps>(
  ({ ...rest }, ref) => <StyledModalContent ref={ref} {...rest} />
);

ModalContent.displayName = "Modal.Content";

export { ModalContent } ;
