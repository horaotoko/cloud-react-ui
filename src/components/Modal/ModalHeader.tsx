import { forwardRef } from "react";
import styled, { css } from "styled-components";
import { defaultProps } from "../../theme";
import { IconButton as MuiIconButton } from "@material-ui/core";

export type ModalHeaderProps = {
  handleClose?: VoidFunction;
  children?: React.ReactNode;
  className?: string;
};

const StyledModalHeader = styled.div`
  ${({ theme: { modal } }) => css`
    display: ${modal.header.display};
    align-items: ${modal.header.alignItems};
    justify-content: ${modal.header.justifyContent};
    background: ${modal.header.background};
    padding: ${modal.header.padding};
    min-height: ${modal.header.minHeight};
    font-size: ${modal.header.fontSize};
    font-weight: ${modal.header.fontWeight};
    word-break: ${modal.header.wordBreak};
  `}
`;
StyledModalHeader.defaultProps = defaultProps;

const CloseButton = styled(MuiIconButton)`
  ${({ theme: { modal } }) => css`
    && {
      flex: ${modal.header.button.flex};
    }
  `}
`;
CloseButton.defaultProps = defaultProps;

const CloseIcon = styled(({ className }) => (
  <svg
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    className={className}
  >
    <title>close</title>
    <path
      d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
      fill="#7C8291"
    />
  </svg>
))`
  ${({ theme: { modal } }) => css`
    width: ${modal.header.button.icon.width};
    height: ${modal.header.button.icon.height};

    & > path {
      fill: ${modal.header.button.icon.color};
    }
  `}
`;
CloseIcon.defaultProps = defaultProps;

const ModalHeader = forwardRef<HTMLDivElement, ModalHeaderProps>(
  ({ children, handleClose, ...rest }, ref) => (
    <StyledModalHeader ref={ref} {...rest}>
      {children}
      <CloseButton size="small" onClick={handleClose}>
        <CloseIcon />
      </CloseButton>
    </StyledModalHeader>
  )
);

ModalHeader.displayName = "Modal.Header";

export { ModalHeader }
