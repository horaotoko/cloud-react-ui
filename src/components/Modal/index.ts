export { Modal } from './Modal';
export type {
  ModalProps,
  ModalHeaderProps,
  ModalContentProps,
  ModalActionsProps,
} from './Modal';
