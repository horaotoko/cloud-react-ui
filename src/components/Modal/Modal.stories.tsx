import { useState } from "react";
import { Story, Meta } from "@storybook/react";
import { Modal } from "./Modal";
import { Button } from "../Button";
import { Text } from "../Text";
export default {
  component: Modal,
  title: "components/Modal",
} as Meta;

const Template: Story = () => {
  const [open, setOpen] = useState(false);
  const  handleClose = () => {
    setOpen(false);
  }

  return (
    <>
      <Button onClick={() => setOpen(!open)}>Show Modal</Button>
      <Modal open={open} maxWidth="sm" fullWidth onClose={handleClose}>
        <Modal.Header handleClose={handleClose}>ModalHeader</Modal.Header>
        <Modal.Content>ModalContent</Modal.Content>
        <Modal.Actions>
          <Button onClick={handleClose}>Cancel</Button>
          <Button color="primary" type="button" >Action</Button>
        </Modal.Actions>
      </Modal>
    </>
  );
};
export const Default = Template.bind({});
