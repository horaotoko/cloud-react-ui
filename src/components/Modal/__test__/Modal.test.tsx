import { useState } from "react"
import { render, screen, fireEvent, waitFor } from "@testing-library/react"
import { Modal } from "../Modal"
import { Button } from "../../Button";

type TestModalProps = {
  initOpenValue: boolean
};

const TestModal = ({ initOpenValue }: TestModalProps): JSX.Element => {
  const [open, setOpen] = useState(initOpenValue);

  return (
    <>
      <Button onClick={() => setOpen(!open)}>Show Modal</Button>
      <Modal open={open} onClose={() => setOpen(false)}>
        <Modal.Header handleClose={() => setOpen(false)}>Header</Modal.Header>
        <Modal.Content>Content</Modal.Content>
        <Modal.Actions>
          <Button>Action</Button>
        </Modal.Actions>
      </Modal>
    </>
  );
};

describe("Modal", () => {
  it("snapshot test", () => {
    const {asFragment} = render(<TestModal initOpenValue={true} />)
    expect(asFragment()).toMatchSnapshot();
  })

  it("opened modal", async () => {
    const {asFragment} = render(<TestModal initOpenValue={false} />)
    
    fireEvent.click(screen.getByText('Show Modal'));

    await waitFor( () => {
      expect(screen.getByRole('dialog')).toBeInTheDocument();
    }) 
    
    expect(asFragment()).toMatchSnapshot();
  })

  it("closed modal with close button", async () => {
    const {asFragment} = render(<TestModal initOpenValue={true}/>)
    
    fireEvent.click(screen.getByLabelText('閉じる'));

    await waitFor( () => {
      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    }) 

    expect(asFragment()).toMatchSnapshot();
  })
})
