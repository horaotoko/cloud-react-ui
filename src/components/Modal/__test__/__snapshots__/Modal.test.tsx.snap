// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal closed modal with close button 1`] = `
<DocumentFragment>
  .c0.c0 {
  display: inline-block;
  height: 32px;
  padding: 0 12px;
  line-height: 32px;
  font-size: 14px;
  border-radius: 4px;
  border: 1px solid #D4D8DD;
  color: #333333;
  background: transparent;
  cursor: pointer;
  position: relative;
  z-index: 0;
}

.c0.c0:before,
.c0.c0:after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  border-radius: 4px;
}

.c0.c0:before {
  z-index: -1;
  background: linear-gradient(to bottom,#FFFFFF,#ECF2FD);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c0.c0:after {
  content: '';
  z-index: -2;
  background: linear-gradient(to bottom,#ECF2FD,#FFFFFF);
  border-radius: 4px;
}

.c0.c0:hover:before {
  opacity: 0;
}

.c0.c0:active,
.c0.c0:focus,
.c0.c0:visited {
  outline: 0;
  box-shadow: 0 0 0 2px rgba(212,216,221,0.3);
}

<button
    class="c0"
  >
    Show Modal
  </button>
</DocumentFragment>
`;

exports[`Modal opened modal 1`] = `
<DocumentFragment>
  .c0.c0 {
  display: inline-block;
  height: 32px;
  padding: 0 12px;
  line-height: 32px;
  font-size: 14px;
  border-radius: 4px;
  border: 1px solid #D4D8DD;
  color: #333333;
  background: transparent;
  cursor: pointer;
  position: relative;
  z-index: 0;
}

.c0.c0:before,
.c0.c0:after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  border-radius: 4px;
}

.c0.c0:before {
  z-index: -1;
  background: linear-gradient(to bottom,#FFFFFF,#ECF2FD);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c0.c0:after {
  content: '';
  z-index: -2;
  background: linear-gradient(to bottom,#ECF2FD,#FFFFFF);
  border-radius: 4px;
}

.c0.c0:hover:before {
  opacity: 0;
}

.c0.c0:active,
.c0.c0:focus,
.c0.c0:visited {
  outline: 0;
  box-shadow: 0 0 0 2px rgba(212,216,221,0.3);
}

<button
    class="c0"
  >
    Show Modal
  </button>
</DocumentFragment>
`;

exports[`Modal snapshot test 1`] = `
<DocumentFragment>
  .c0.c0 {
  display: inline-block;
  height: 32px;
  padding: 0 12px;
  line-height: 32px;
  font-size: 14px;
  border-radius: 4px;
  border: 1px solid #D4D8DD;
  color: #333333;
  background: transparent;
  cursor: pointer;
  position: relative;
  z-index: 0;
}

.c0.c0:before,
.c0.c0:after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  border-radius: 4px;
}

.c0.c0:before {
  z-index: -1;
  background: linear-gradient(to bottom,#FFFFFF,#ECF2FD);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c0.c0:after {
  content: '';
  z-index: -2;
  background: linear-gradient(to bottom,#ECF2FD,#FFFFFF);
  border-radius: 4px;
}

.c0.c0:hover:before {
  opacity: 0;
}

.c0.c0:active,
.c0.c0:focus,
.c0.c0:visited {
  outline: 0;
  box-shadow: 0 0 0 2px rgba(212,216,221,0.3);
}

<button
    class="c0"
  >
    Show Modal
  </button>
</DocumentFragment>
`;
